genie-core/src/main/java/com/netflix/genie/core/jobs/workflow/impl/JobFailureAndKillHandlerLogicTask.java

```diff
@@ -39,15 +40,16 @@
 @Slf4j
 public class JobFailureAndKillHandlerLogicTask extends GenieBaseTask {
 
-    private final Timer timer;
+    private final Id timerId;
 
     /**
      * Constructor.
      *
      * @param registry The metrics registry to use
      */
     public JobFailureAndKillHandlerLogicTask(@NotNull final Registry registry) {
-        this.timer = registry.timer("genie.jobs.tasks.jobFailureAndKillHandlerLogicTask.timer");
+        super(registry);
+        this.timerId = registry.createId("genie.jobs.tasks.jobFailureAndKillHandlerLogicTask.timer");
```

Comment:
Or you might just want to add Utility methods to the base class or metrics utils? I mean the Registry is one global registry so perhaps something like a RegistryUtils we pass around? Dunno.